### PR TITLE
Remove redundant pull docker image

### PR DESCRIPTION
**Pull request by sima-zhu**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
The PR is to remove redundant step of pull docker image.

image

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 1]
```diff
@@ -235,11 +235,6 @@ stages:
                         displayName: 'Download Artifact: ${{parameters.ArtifactName}}'
                         artifact: ${{parameters.ArtifactName}}
                       # Pull and build the docker image.
-                      - template: /eng/common/pipelines/templates/steps/docker-pull-image.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release.yml:238]: Where is the other pull happening?

Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-java-release.yml:238]: Do we need to clean up any of these variables or are they still used for the other pull?

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-java-release.yml:238]: It is inside of update-docsms-metadata.yml

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-java-release.yml:238]: https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/pipelines/templates/steps/update-docsms-metadata.yml#L82

**Pull request merged**